src/eventHubMessageExplorer.ts

```diff
@@ -0,0 +1,99 @@
+    private _eventHubPathKey = "eventHubPath";
+    private _eventHubPathTitle = "Event Hub Path";
+    private _monitorLabel = "MonitorEventHub";
+    private _messageLabel = 'MessageToEventHub';
+
+    public sendMessageToEventHub(): void {
+        let eventHubConnectionString = Utility.getConfig(this._eventHubConnectionstringKey, this._eventHubConnectionStringTitle);
+        let eventHubPath = Utility.getConfig(this._eventHubPathKey, this._eventHubPathTitle);
+        if(!eventHubConnectionString || !eventHubPath) {
+            this.outputLine(this._messageLabel, 'Invalid arguments of event hub.');
+        }
+        vscode.window.showInputBox({ prompt: 'Enter message to send to event hub' }).then((message: string) => {
+            if (message !== undefined) {
+                try {
+                    let client = EventHubClient.fromConnectionString(eventHubConnectionString, eventHubPath);
+                    let receiveAfterTime = Date.now() - 5000;
+                    client.open()
+                        .then(client.getPartitionIds.bind(client))
+                        .then(() => client.createSender())
+                        .then(this.sendToEventHub(message, this._messageLabel))
+                        .then(this.sendToEventHubDone(client, this._messageLabel))
+                        .catch(this.printError(this._outputChannel, this._messageLabel, client));
+                }
+                catch (e) {
+                    this.outputLine(this._messageLabel, e);
+                }
+            }
+        });
+    }
+
+    public startMonitorEventHubMessage(): void {
+        let eventHubConnectionString = Utility.getConfig(this._eventHubConnectionstringKey, this._eventHubConnectionStringTitle);
+        let eventHubPath = Utility.getConfig(this._eventHubPathKey, this._eventHubPathTitle);
+        if(!eventHubConnectionString || !eventHubPath) {
+            this.outputLine(this._monitorLabel, 'Invalid arguments of event hub!');
+        }
+        try {
+            this._eventHubClient = EventHubClient.fromConnectionString(eventHubConnectionString, eventHubPath);
+            let receiveAfterTime = Date.now() - 5000;
+            this._outputChannel.show();
+            this.outputLine(this._monitorLabel, 'Start monitor event hub ...');
+            this._appInsightsClient.sendEvent('EventHub.startMonitor');
+            this._eventHubClient.open()
+                .then(this._eventHubClient.getPartitionIds.bind(this._eventHubClient))
+                .then((partitionIds: EventHubClient.PartitionId[]) =>
+                    Promise.map(partitionIds, (partitionId) =>
+                        this._eventHubClient.createReceiver('$Default', partitionId, { startAfterTime: receiveAfterTime })
+                            .then((receiver: EventHubReceiver) => {
+                                receiver.on('errorReceived', this.printError(this._outputChannel, this._monitorLabel, this._eventHubClient));

```

Comment:
Better to add output when receiver is created. Refer to https://github.com/formulahendry/vscode-azure-iot-toolkit/blob/master/src/messageExplorer.ts#L57